emcache/client.py

```diff
@@ -647,6 +651,79 @@ async def version(self, memcached_host_address: MemcachedHostAddress) -> Optiona
         _, version_value = result.split(b" ")
         return version_value.decode()
 
+    async def gat(self, key: bytes, exptime: int = 0, return_flags=False) -> Optional[Item]:
+        """Gat command is used to fetch item and update the
+        expiration time of an existing item.
+
+        gat <exptime> <key>\r\n
+        """
+        keys, values, flags, _ = await self._fetch_command(b"gat %d" % exptime, key)
```

Comment:
reusing the `_fetch_command` and adding as a command the interpolation of the command and the exptime seems to twist a bit the interfaces. The other option would to also create a two new code paths for the fetch commands that require the `gat`(s)  commands. Yes I know will imply add more code but defenitly would look nicer :)

What do you think?

Reply:
thanks for the note, it’s really better not to mix everything into one team

Reply:
used new method _get_and_touch_command with required arg exptime